### DISCUSSION TITLE
Improve usability of working with lots of index.vue files

### DISCUSSION THREAD

**Discussion by mariusa** [hannoeru/vite-plugin-pages#181] [Ideas]
Hi,

vite-plugin-pages works great. When editing code though (eg with vscode), it's hard to navigate through lots of `index.vue` files. eg I want to open the products listing, or contacts listing. All of these are named like `products/index.vue`

As a suggestion, it would be useful to have `products/products.vue` be served exactly as `products/index.vue` (when filename matches parent folder). This way we can get rid of `index.vue`'s and still keep the proper routes & folders organization.

What do you think? How could this be done?

Thanks!

**Comment by hannoeru** [comment]
I don't think this is a good idea, what if user want to use `/products/products` route? they will need to find another way to do that.
But you can use `extendRoute` to rewrite the route path, I think that is enough for your requirement.

**Reply by mariusa** [reply]
Thank you! `extendRoute`  did the job :)
